crates/context/src/lib.rs

```diff
@@ -748,6 +758,28 @@ impl ContextManager {
         Ok(ids)
     }
 
+    pub fn get_own_signing_key(
+        &self,
+        context_id: &ContextId,
+        own_public_key: &PublicKey,
+    ) -> EyreResult<SigningKey> {
+        let handle = self.store.handle();
+        let key = handle
+            .get(&ContextIdentityKey::new(*context_id, *own_public_key))?
+            .unwrap() // we are sure it exists because we own the public_key
```

Comment:
it won't always exist, because we won't only want to fetch the sender key for our identity